gensim/corpora/textcorpus.py

```diff
@@ -191,19 +203,34 @@ def __init__(self, source=None, dictionary=None, metadata=False, character_filte
                 will be built for the given corpus. If no corpus is given, the dictionary will
                 remain uninitialized.
             metadata (bool): True to yield metadata with each document, else False (default).
+                By default, metadata consists only of the line number for each document. This
+                is yielded from `get_texts` as a tuple of (tokens, (line_num,)) and from __iter__
+                as a tuple of (bag_of_words, (line_num,)).
             character_filters (iterable of callable): each will be applied to the text of each
                 document in order, and should return a single string with the modified text.
-                For Python 2, the original text will not be unicode, so it may be useful to
-                convert to unicode as the first character filter. The default character filters
-                lowercase, convert to unicode (strict utf8), perform ASCII-folding, then collapse
+                For Python 2, the original text will not be unicode (unless you modify your
+                `getstream` method to convert it to unicode), so it may be useful to convert to 
+                unicode as the first character filter. The default character filters lowercase, 
+                convert to unicode (strict utf8), perform ASCII-folding, then collapse
                 multiple whitespaces.
             tokenizer (callable): takes as input the document text, preprocessed by all filters
-                in `character_filters`; should return an iterable of tokens (strings).
+                in `character_filters`; should return an iterable of tokens. The tokens should
+                be in a format that can be parsed by the first callable in `token_filters`.
             token_filters (iterable of callable): each will be applied to the iterable of tokens
                 in order, and should return another iterable of tokens. These filters can add,
                 remove, or replace tokens, or do nothing at all. The default token filters
                 remove tokens less than 3 characters long and remove stopwords using the list
                 in `gensim.parsing.preprocessing.STOPWORDS`.
+            processes (int): number of processes to use for text preprocessing. The default is
+                -1, which will use (number of virtual CPUs - 1) worker processes, in addition
+                to the master process. If set to a number greater than the number of virtual
+                CPUs available, the value will be reduced to (number of virtual CPUs - 1).
```

Comment:
you're right; this mistrust of users is not suitable for a Python code base! Modified to remove upper bounding